nexus/src/app/background/init.rs

```diff
@@ -2,7 +2,90 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-//! Specific background task initialization
+//! Initialize Nexus background tasks
+//!
+//! This file contains entirely Nexus-specific initialization (as opposed to
+//! driver.rs, which doesn't really know much about Nexus).
+//!
+//! The design here is oriented around being able to initialize background tasks
+//! in two phases:
+//!
+//! 1. Phase 1 assembles a `BackgroundTasks` struct containing `Activator`
+//!    objects that will be used by any part of Nexus (including background
+//!    tasks) to activate any background task or read data provided by another
+//!    background task.  This is the interface between this subsystem and the
+//!    rest of Nexus.  At this point in startup, none of the background tasks
+//!    themselves have been started yet.
+//!
+//! 2. Phase 2 starts all of the individual background tasks and then wires up
+//!    the Activators created in phase 1.
+//!
+//! This allows us to break what would otherwise be a circular dependency during
+//! initialization.  Concretely: Nexus startup does phase 1, stores the
+//! `BackgroundTasks` into the `Arc<Nexus>` to which all of Nexus has a
+//! reference, and _then_ starts the background tasks.  If we didn't break it up
+//! like this, then we couldn't make the `Arc<Nexus>` available to background
+//! tasks during _their_ initialization (because it couldn't be constructed
+//! yet), which means background tasks could not activate other background
+//! tasks.  We'd also have trouble allowing background tasks to use other
+//! subsystems in Nexus (e.g., sagas), especially if those subsystems wanted to
+//! activate background tasks.
+//!
+//! Why do we do things this way?  We're trying to satisfy a few different
+//! goals:
+//!
+//! - Background tasks should be able to activate other background tasks.
+//! - Background tasks should be able to use other subsystems in Nexus (like
+//!   sagas) that themselves can activate background tasks.
+//! - It should be hard to mess any of this up when adding or removing
+//!   background tasks.  This means:
+//!     - We should be able to tell at compile-time which code activates what
+//!       specific background tasks.
+//!     - We should be able to tell at compile-time if code is attempting to
+//!       activate a background task that doesn't exist.
+//!     - It should be hard to add an Activator for a background task that is
+//!       not wired up to that task or is wired up to a different task.
```

Comment:
this is excellent, thank you for including a list of goals in the docs!

Reply:
I will admit that I'm a little bit uncertain that preventing an activator from being wired up to multiple tasks is all that important. Maybe there are legitimate reasons we might want to have a set of background tasks with different timer periods that are guaranteed to always be explicitly activated as a group, so I don't know if having an activator point at multiple tasks is _always_ a bug? Off the top of my head, I dunno if I have an obvious example use case, but it doesn't seem totally inconceivable.

On the other hand, an `Activator` that _doesn't_ point at any background task after the tasks have been initialized is obviously a bug and we should definitely stop that from happening.

Reply:
Yeah, there's nothing intrinsically wrong with like a `MultiTaskActivator` or something (and it could be the same type if we want).  I made this really strict because based on how we use this today, if an Activator was wired up to two tasks, that almost certainly means somebody copied/pasted a `Driver::register()` call and forgot to change that argument.  I really want to avoid this, since the failure mode is the worst kind of implicit, non-fatal breakage: things mostly work, but they suck, and the problem will be pretty tough to track down.